streaming/src/main/scala/io/gearpump/streaming/task/TaskActor.scala

```diff
@@ -242,6 +243,16 @@ class TaskActor(
       receiveMessage(inputMessage, sender)
     case upstream@ UpstreamMinClock(upstreamClock) =>
       this.upstreamMinClock = upstreamClock
+      val curMin = minClockAtCurrentTask
+      if (curMin == curTaskMinClock) {
+        subscriptions.foreach { case (_, subscription) =>
```

Comment:
This is the hot path, maybe using while is better(foreach is not natively support by java)